wasm/src/main/java/com/dylibso/chicory/wasm/types/CodeSection.java

```diff
@@ -1,14 +1,22 @@
 package com.dylibso.chicory.wasm.types;
 
 import java.util.ArrayList;
-import java.util.List;
+import java.util.Objects;
 
 public class CodeSection extends Section {
     private final ArrayList<FunctionBody> functionBodies;
 
-    public CodeSection(FunctionBody[] functionBodies) {
+    private CodeSection(ArrayList<FunctionBody> functionBodies) {
         super(SectionId.CODE);
-        this.functionBodies = new ArrayList<>(List.of(functionBodies));
+        this.functionBodies = functionBodies;
```

Comment:
If the builder is used after the `build` call, the lists of everything that was built with it will be modified. You'd need to make a copy here e.g. `List.copyOf(functionBodies)`.